### PR TITLE
Get back on Rust stable [ci full]

### LINKED ISSUE #2201: Find a way to come back on Rust stable
We're [pinned](https://github.com/mozilla/application-services/pull/2171) on Rust 1.38.0 until we figure out why `cc` is failing cross-compilation on Windows with Rust 1.39.0.
Bad run example:
https://firefox-ci-tc.services.mozilla.com/tasks/groups/SNh1AyfBQPePaleBDlWImA

Removing `backtrace` does not help because `cc` is used by other crates.

┆Issue is synchronized with this [Jira Story](https://jira.mozilla.com/browse/SYNC-394)
┆Story Points: 5
┆Epic: important not urgent
┆Sprint: SYNC - end 2020-01-03

### PR DESCRIPTION
**Pull request by eoger**
Fixes #2201.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -242,27 +242,26 @@ jobs:
...
-          rust-version: 1.38.0
+      - test-setup
       - run: rustup component add clippy
       - run: cargo clippy --version
-      - run: cargo clippy --all --all-targets --all-features -- -D warnings
+      - run: cargo clippy --all --all-targets --all-features -- -D warnings -A clippy::redundant_clone
       # Run clippy without any features so that we still lint things that are
       # behind a `#[cfg(not(feature = "..."))]`, which will always be false with
       # --all-features.
-      - run: cargo clippy --all --all-targets --no-default-features -- -D warnings
```
Comment (eoger) [inline .circleci/config.yml]: This started failing with Rust 1.40.0, I can open a follow-up to see if we can do anything about this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -242,27 +242,26 @@ jobs:
...
-      - image: circleci/rust:1.38
+      - image: circleci/rust:latest
     steps:
-      - test-setup:
-          rust-version: 1.38.0
+      - test-setup
       - run: rustup component add clippy
       - run: cargo clippy --version
-      - run: cargo clippy --all --all-targets --all-features -- -D warnings
+      - run: cargo clippy --all --all-targets --all-features -- -D warnings -A clippy::redundant_clone
```
Comment (eoger) [inline .circleci/config.yml]: I'm believe I'm seeing false-positives from this lint, but maybe I'm not understanding the semantics of the code. In any case this should not block this PR from landing.

Comment (thomcc) [reply .circleci/config.yml]: Can you (have you) filed a followup issue?

File: .circleci/config.yml (hunk 1 above)
Comment (thomcc) [reply .circleci/config.yml]: See https://github.com/mozilla/application-services/blob/ffb8f3cd87e24e0f8fdf4b48af10a6dd69bda48d/.circleci/config.yml#L148-L163, where we worked around this elsewhere.

**Pull request merged**